src/Nethermind/Nethermind.JsonRpc/Modules/Rbuilder/RbuilderRpcModule.cs

```diff
@@ -0,0 +1,162 @@
+                    // Set, its either this or changing `IWorldState` which is somewhat risky.
+                    if (accountChange.Nonce is not null)
+                    {
+                        worldState.SetNonce(address, accountChange.Nonce.Value);
+                    }
+
+                    if (accountChange.Balance is not null)
+                    {
+                        UInt256 originalBalance = account.Balance;
+                        if (accountChange.Balance.Value > originalBalance)
+                        {
+                            worldState.AddToBalance(address, accountChange.Balance.Value - originalBalance, releaseSpec);
+                        }
+                        else if (accountChange.Balance.Value == originalBalance)
+                        {
+                        }
+                        else
+                        {
+                            worldState.SubtractFromBalance(address, originalBalance - accountChange.Balance.Value, releaseSpec);
+                        }
+
+                        if (worldState.GetBalance(address) != accountChange.Balance.Value)
+                        {
+                            throw new Exception("Balance is not same!? Why?");
+                        }
+                    }
+                }
+                else
+                {
+                    worldState.CreateAccountIfNotExists(address, accountChange.Balance ?? 0, accountChange.Nonce ?? 0);
+                }
+
+                if (accountChange.CodeHash is not null)
+                {
+                    // Note, this set CodeDb, but since this is a read only world state, it should do nothing.
+                    worldState.InsertCode(address, accountChange.CodeHash, Array.Empty<byte>(), releaseSpec, false);
+                }
+
+                if (accountChange.ChangedSlots is not null)
+                {
+                    foreach (KeyValuePair<UInt256, UInt256> changedSlot in accountChange.ChangedSlots)
+                    {
+                        ReadOnlySpan<byte> bytes = changedSlot.Value.ToBigEndian();
+                        bool newIsZero = bytes.IsZero();
+                        bytes = !newIsZero ? bytes.WithoutLeadingZeros() : [0];
+                        worldState.Set(new StorageCell(address, changedSlot.Key), bytes.ToArray());
+                    }
+                }
+            }
+
+            worldState.Commit(releaseSpec);
+            worldState.CommitTree(blockHeader.Number + 1);
+            return ResultWrapper<Hash256>.Success(worldState.StateRoot);
+        }
+        finally
+        {
+            _overridableWorldScopePool.Return(worldScope);
+        }
+    }
+
+
+    public ResultWrapper<AccountState> rbuilder_getAccount(Address address, BlockParameter block)
+    {
+        BlockHeader? blockHeader = blockFinder.FindHeader(block);
+        if (blockHeader is null)
+        {
+            return ResultWrapper<AccountState>.Fail("Block not found");
```

Comment:
```suggestion
            return ResultWrapper<Hash256>.Fail("Block not found", ErrorCodes.ResourceNotFound);
```

Reply:
Maybe there should be some helper methods in `ResultWrapper` for most used error types?